### PR TITLE
Upmerge to the latest CUDF and fix compile errors

### PR DESCRIPTION
**Pull request by revans2**
This upmerges to the latest 23.06 CUDF and fixes the compile errors in our code that showed up with it.

I will be testing this on our plugin too, but I wanted to get this up and started to unblock people.

### CODE REVIEW

**Comment by revans2** [comment]
build

File: src/main/cpp/src/cast_string.cu
Comment (jlowe) [inline src/main/cpp/src/cast_string.cu:1]: 2023 copyrights

**Comment by revans2** [comment]
Just FYI there was a breaking change made in some of the CUDF java code that I will also put up a patch in the plugin to fix.

**Comment by revans2** [comment]
https://github.com/NVIDIA/spark-rapids/pull/8328 will fix the issues in the plugin. I am going to run more tests still. I mostly just got it to compile before putting up the PR...

**Comment by revans2** [comment]
build

**Comment by abellina** [comment]
Ran our jni tests, things seem ok from that perspective.

**Review by jlowe — APPROVED** [review]

**Comment by revans2** [comment]
oops didn't fix the test code :( doing that now...

**Comment by revans2** [comment]
build

**Review by jlowe — APPROVED** [review]

**Review by abellina — APPROVED** [review]

**Pull request merged**